src/sentry/workflow_engine/typings/action/types.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from sentry.incidents.models.alert_rule import (
+    AlertRule,
+    AlertRuleDetectionType,
+    AlertRuleThresholdType,
+    AlertRuleTriggerAction,
+)
+from sentry.issues.issue_occurrence import IssueOccurrence
+from sentry.workflow_engine.models import Action, Detector
+
+
+@dataclass
+class NotificationContext:
+    """
+    NotificationContext is a dataclass that represents the context required send a notification.
+    """
+
+    integration_id: int | None = None
+    target_identifier: str | None = None
+    target_display: str | None = None
+    sentry_app_id: str | None = None
+    sentry_app_config: list[dict[str, Any]] | dict[str, Any] | None = None
+
+    @classmethod
+    def from_alert_rule_trigger_action(cls, action: AlertRuleTriggerAction) -> NotificationContext:
+        return cls(
+            integration_id=action.integration_id,
+            target_identifier=action.target_identifier,
+            target_display=action.target_display,
+            sentry_app_config=action.sentry_app_config,
+        )
+
+    @classmethod
+    def from_action_model(cls, action: Action) -> NotificationContext:
+        if action.type == Action.Type.SENTRY_APP:
+            return cls(
+                integration_id=action.integration_id,
+                target_display=action.config.get("target_display"),
+                sentry_app_config=action.data.get("settings"),
+                sentry_app_id=action.data.get("target_identifier"),
+                # Does not need data
+            )
+        elif action.type == Action.Type.OPSGENIE or action.type == Action.Type.PAGERDUTY:
+            return cls(
+                integration_id=action.integration_id,
+                target_identifier=action.config.get("target_identifier"),
+                target_display=action.config.get("target_display"),
+                sentry_app_config=action.data,
+            )
+        # TODO(iamrajjoshi): Add support for email here
+
+        return cls(
+            integration_id=action.integration_id,
+            target_identifier=action.config.get("target_identifier"),
+            target_display=action.config.get("target_display"),
+        )
+
+
+@dataclass
+class AlertContext:
```

Comment:
I don't think that this class belongs in the workflow engine - we're using specific imports from the `incidents` module which is a bit of a code smell to me.

Is there a centralized place where we are putting all the technical debt from metric alerts registry? Maybe like `metric_alert.py` file? Ideally we co-locate _all_ technical debt that is caused by these registries - by having this code in the base typings info, it will likely be harder for us to track down. (It seems like we should also only be doing this when invoking methods for the metric alert registry)